sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -966,6 +966,9 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
       case _: FlatMapGroupsInPandasWithState =>
         // TODO(SPARK-40443): support applyInPandasWithState in batch query
         throw new SparkUnsupportedOperationException("_LEGACY_ERROR_TEMP_3176")
+      case _: TransformWithStateInPandas =>
+        // TODO(SPARK-50428): support TransformWithStateInPandas in batch query
+        throw new SparkUnsupportedOperationException("_LEGACY_ERROR_TEMP_3176")
```

Comment:
nit: sorry you need to define another error class; it's saying applyInPandasWithState. OK to add another temporary error class as we want to fix it anyway.

Reply:
Reuse this error class used for `TransformWithState` previously: [https://github.com/apache/spark/commit/289543eba5bd07395282c28cb08934ec625a4935#diff-7c879c08d2f379c139d5229a888[…]69bb48f0a138a3d64e1b2dde3502feR46](https://github.com/apache/spark/commit/289543eba5bd07395282c28cb08934ec625a4935#diff-7c879c08d2f379c139d5229a88857229ae69bb48f0a138a3d64e1b2dde3502feR46)